libraries/DataFlash/LogFile.cpp

```diff
@@ -1924,21 +1924,33 @@ void DataFlash_Class::Log_Write_Current(const AP_BattMonitor &battery)
         struct log_Current pkt = {
             LOG_PACKET_HEADER_INIT(LOG_CURRENT_MSG),
             time_us             : AP_HAL::micros64(),
-            battery_voltage     : battery.voltage(0),
+            voltage             : battery.voltage(0),
+            voltage_resting     : battery.voltage_resting_estimate(0),
             current_amps        : battery.current_amps(0),
             current_total       : battery.current_total_mah(0),
             temperature         : (int16_t)(has_temp ? (temp * 100) : 0),
+            resistance          : battery.get_resistance(0)
         };
-        AP_BattMonitor::cells cells = battery.get_cell_voltages(0);
+        WriteBlock(&pkt, sizeof(pkt));
 
         // check battery structure can hold all cells
-        static_assert(ARRAY_SIZE(cells.cells) == (sizeof(pkt.cell_voltages) / sizeof(pkt.cell_voltages[0])),
+        struct log_Current_Cells cell_pkt;
+        const AP_BattMonitor::cells &cells = battery.get_cell_voltages(0);
+        static_assert(ARRAY_SIZE(cells.cells) == (sizeof(cell_pkt.cell_voltages) / sizeof(cell_pkt.cell_voltages[0])),
```

Comment:
I wasn't sure if the static_assert applied at compile time or runtime.. others tell me it's compile time so I will move it as suggested.